dev/archery/archery/linking.py

```diff
@@ -61,9 +64,71 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _extract_symbols(self, symbol_info):
+        return [re.search(r'\S+$', line).group() for line in symbol_info if line]
+
+    def _remove_weak_symbols(self, lines):
+        return [line for line in lines if not re.search(r'\s[Ww]\s', line)]
+
+    def list_symbols_for_dependency(self, dependency, remove_symbol_versions=False):
+        result = _nm.run('-D', dependency, stdout=subprocess.PIPE)
+        lines = result.stdout.decode('utf-8').splitlines()
+        if remove_symbol_versions:
+            lines = [re.split('@@', line)[0] for line in lines]
```

Comment:
How about extracting this as a method because this is used in `list_undefined_symbols_for_dependency()` too?